From 66faad965df6df47660b8b14e54823bfa010a5bd Mon Sep 17 00:00:00 2001
From: Ratheesh Kannoth <rkannoth@codeaurora.org>
Date: Thu, 4 Jun 2020 11:39:22 +0530
Subject: [PATCH] net: bonding: Relocate kernel bond functions

header files included in uapi folder are the interface between
userspace and kernel space. This path declares new bonding
functions in kernel header file instead of uapi header file.

Signed-off-by: Ratheesh Kannoth <rkannoth@codeaurora.org>
Change-Id: I1b33f07015b79ec4ac702a9079da01f1c2a08bbb
---
 include/net/bonding.h           | 13 +++++++++++++
 include/uapi/linux/if_bonding.h | 14 --------------
 2 files changed, 13 insertions(+), 14 deletions(-)

--- a/include/net/bonding.h
+++ b/include/net/bonding.h
@@ -754,4 +754,17 @@ static inline netdev_tx_t bond_tx_drop(s
 	return NET_XMIT_DROP;
 }
 
+struct bond_cb {
+	void (*bond_cb_link_up)(struct net_device *slave);
+	void (*bond_cb_link_down)(struct net_device *slave);
+	void (*bond_cb_enslave)(struct net_device *slave);
+	void (*bond_cb_release)(struct net_device *slave);
+	void (*bond_cb_delete_by_slave)(struct net_device *slave);
+	void (*bond_cb_delete_by_mac)(uint8_t *mac_addr);
+};
+
+extern int bond_register_cb(struct bond_cb *cb);
+extern void bond_unregister_cb(void);
+extern int bond_get_id(struct net_device *bond_dev);
+
 #endif /* _NET_BONDING_H */
--- a/include/uapi/linux/if_bonding.h
+++ b/include/uapi/linux/if_bonding.h
@@ -152,19 +152,4 @@ enum {
 };
 #define BOND_3AD_STAT_MAX (__BOND_3AD_STAT_MAX - 1)
 
-#ifdef __KERNEL__
-struct bond_cb {
-	void (*bond_cb_link_up)(struct net_device *slave);
-	void (*bond_cb_link_down)(struct net_device *slave);
-	void (*bond_cb_enslave)(struct net_device *slave);
-	void (*bond_cb_release)(struct net_device *slave);
-	void (*bond_cb_delete_by_slave)(struct net_device *slave);
-	void (*bond_cb_delete_by_mac)(uint8_t *mac_addr);
-};
-
-extern int bond_register_cb(struct bond_cb *cb);
-extern void bond_unregister_cb(void);
-extern int bond_get_id(struct net_device *bond_dev);
-#endif /* __KERNEL__ */
-
 #endif /* _LINUX_IF_BONDING_H */
